hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/HeapData.java

```diff
@@ -162,6 +162,19 @@ public boolean isPortable() {
         return SerializationConstants.CONSTANT_TYPE_PORTABLE == getType();
     }
 
+    /**
+     * Converts Data to HeapData. Useful for offheap conversion.
+     *
+     * @param data
+     * @return the onheap representation of data. If data is null, null is returned.
+     */
+    public static Data toOnHeap(Data data) {
```

Comment:
It would be best to model this as a instance method instead of a static method.

So you can always say 'data.toHeap' no matter the type of data. Current approach is more procedural instead of object oriented.